api/app/c_haines/severity_index.py

```diff
@@ -0,0 +1,453 @@
+""" Logic pertaining to the generation of c_haines severity index from GDAL datasets.
+"""
+import os
+from datetime import datetime, timezone, timedelta
+from typing import Final, Tuple, Generator, Union, List
+from contextlib import contextmanager
+import tempfile
+import logging
+import json
+from osgeo import gdal
+import ogr
+import numpy
+from pyproj import Transformer, Proj
+from shapely.ops import transform
+from shapely.geometry import shape
+from sqlalchemy.orm import Session
+from app.time_utils import get_utc_now
+from app.db.models.c_haines import CHainesPoly, CHainesPrediction, CHainesModelRun, severity_levels
+from app.db.crud.weather_models import get_prediction_model
+from app.db.crud.c_haines import (get_c_haines_prediction, get_or_create_c_haines_model_run)
+from app.weather_models import ModelEnum, ProjectionEnum
+from app.weather_models.process_grib import NAD83
+from app.weather_models.env_canada import (get_model_run_hours,
+                                           get_file_date_part, adjust_model_day, download,
+                                           UnhandledPredictionModelType)
+from app.c_haines.c_haines_index import CHainesGenerator
+from app.c_haines import GDALData
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_severity(c_haines_index) -> int:
+    """ Return the "severity" of the continuous haines index.
+
+    Fire behavrious analysts are typically only concerned if there's a high
+    or extreme index - so the c-haines values are lumped together by severity.
+
+    The severity used here is fairly arbitrary - there's no standard in place.
+    """
+    # 0 - 4 : low
+    if c_haines_index < 4:
+        return 0
+    # 4 - 8 : moderate
+    if c_haines_index < 8:
+        return 1
+    # 8 - 11 : high
+    if c_haines_index < 11:
+        return 2
+    # 11 + Extreme
+    return 3
+
+
+def get_severity_string(severity: int) -> str:
+    """ Return the severity level as a string, e.g. severity level 3 maps to "11+"
```

Comment:
I was thinking of changing the method to accept a `SeverityEnum` so we could just reuse that instead of switching on integers, but then I followed the call chain up to see how the c-haines data is evaluated and it only knows numbers so yeah it's a bit trickier -- feel free to disregard!